tests/asgi/test_asgi.py

```diff
@@ -0,0 +1,217 @@
+import asyncio
+from collections import Counter
+from io import StringIO
+import logging
+import multiprocessing
+import os
+import sys
+import time
+
+import pytest
+import requests
+import uvicorn
+
+import falcon
+import falcon.asgi
+import falcon.testing as testing
+
+_SERVER_HOST = '127.0.0.1'
+_SERVER_PORT = 9000 + os.getpid() % 100  # Facilitates parallel test execution
```

Comment:
Note that while OS PIDs are normally assigned sequentially on Unix-like systems, one needs to walk around existing PIDs for other programs, furthermore, allocation is restarted (often to PID 300) at some point to recycle IDs. The approach presented here is probably good enough for practical use, but it be worth adding some disclaimer note?

Reply:
I went ahead and just switched to using a simple RNG with a fairly large range. That should be sufficient and be easier to reason about.